assets/src/edit-story/app/history/useHistoryReducer.js

```diff
@@ -50,14 +57,30 @@ const reducer = (size) => (state, { type, payload }) => {
       // and clear `offset` and `replayState`.
       return {
         entries: [payload, ...state.entries.slice(state.offset)].slice(0, size),
+        globalHistoryLength: state.globalHistoryLength + 1,
+        hasHistoryChanged: true,
```

Comment:
Why do we need to have both `hasHistoryChanged` and `globalHistoyLength`?

Reply:
`hasHistoryChanged` is just `globalHistoyLength > 0`

Reply:
Actually not, `hasHistoryChanged` allows to control a temporary scope of history changes, avoiding race conditions registering `beforeunload` event. For this scenario, only a counter is not enough, we also should consider that our system update the versionNumber systematically without user intention (e.g. creating a new story, uploading files, etc)